src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -248,36 +250,68 @@ def validate_paths(self, data, schema_name, verrors):
             verrors.add(f"{schema_name}.alldirs", "This option can only be used for datasets")
 
     @private
-    def validate_user_networks(self, other_shares, data, schema_name, verrors):
+    def validate_hosts_and_networks(self, other_shares, data, schema_name, verrors):
         dev = os.stat(data["paths"][0]).st_dev
 
-        used_networks = []
+        used_networks = defaultdict(lambda: 0)
         for share in other_shares:
             try:
                 share_dev = os.stat(share["paths"][0]).st_dev
             except Exception:
                 self.logger.warning("Failed to stat first path for %r", share, exc_info=True)
                 continue
 
-            used_networks.extend([(network, share_dev) for network in share["networks"]])
+            if share_dev == dev:
+                for host in share["hosts"]:
+                    try:
+                        host = socket.gethostbyname(host)
```

Comment:
Resolving hostnames might take a good while, I would think we need to resolve them in a batch and cache it.

We have `resolve_hostname` async validator.